src/gui/iconutils.cpp

```diff
@@ -1,52 +1,140 @@
+/*
+ * Copyright (C) by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
 #include "iconutils.h"
 
 #include <theme.h>
 
 #include <QFile>
+#include <QLoggingCategory>
 #include <QPainter>
 #include <QPixmapCache>
 #include <QSvgRenderer>
 
+namespace {
+QString findSvgFilePath(const QString &fileName, const QStringList &possibleColors)
```

Comment:
better use the Return Value Optimization
so single declared variable and multiple return instructions

Reply:
I think the compiler should be able to optimize that away anyway https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIMwCspK4AMngMmAByPgBGmMQgkgDMpAAOqAqETgwe3r7%2BQemZjgJhEdEscQnJtpj2JQxCBEzEBLk%2BfoG19dlNLQRlUbHxiSkKza3t%2BV3j/YMVVaMAlLaoXsTI7BwA9NsA1AAqAJ6pmHtHa8R7aFh7CPGYpHske7SoTOh7hnuYqqyp9AAdCYNABBEHgsxJcLIby3ExJNwAN0wDhICOwEPM0IYsK88MRTnGxEwrAxWKhMLhZwRbjYLBIR3JYOxVPxNMRyHG6FoeBizMhOLxBLcRIIJLJSUxLMpuOpe1pxPCwAFEOJXgcewA6sQmKlTlcsQB2Kxg1JeGK85AgCF7O17bkgEBKowO8UAfQRpvBYPt2t1%2BviECWCqSABE9lgqEwvLQCF6sb77b8AXhkIR/XqDRBHc7xcq3cQQyBC%2B6cwR0E6GSjy0Wlrb7SYTQ2/Xbc2gvARQ25aQqzGY0cRzGZu73c65aAmk3am2HE6C/Tqs0Gl4Gh2YAGzmDfPAj3It7EvEsu56uYCCoPfxQHHpb16cK5sPv3ttZd2m94dn4ejxGHo8Vk6E5TgujZGnOLKQaBdqrtmaAMOMmZrtuu77sWpYXlexA3h697QY%2B3qtvar6dr%2Bbh9gOqCpEy/Zkf%2BbqViAwFJIRYEQeC4EgRC7rukwBD5jEnaYDxUAMKg4S8hEd4QrB8R7CwTDAGm7pUF4uINBA4RdngeFNqxMEBgaDqoGwslDixLbGaZhnXseoYRrmRDui6wCafWFkPngVB7Jp3Z7CO24oQoJmYGZOHEO69kIg5gEgE5LluSGemWX6JIEOsDBWaFNnmfpj7sX6dRKHsXk%2BXgfkBElT74YuOV7AA7jlIFEXajXLthdnRRRACyeDoDymDDs1LVpRlDVNR5%2BGzpZo3EJlZnJf2Fh4Cq/bTZN01QRCWnyUw4TBsaeXwYhMZEHsuoMOgJnugwwyGuG52GOgB2TX6x1dqdqDje19m7UpyAqWpDjZBAF1XSwN13e5eUke%2BiK9m1a7hZFH5/uOl2Tht4EcCstCcAEvB%2BBwWikKgnA9pY1jGesmx9kkPCkAQmg4ysADWIBJEkgKczzvN8xu%2BicJIhPM6TnC8AoIAaIzzMrHAsBIGgLCpHQ8TkJQSsq/QCTAFwXBmHwdAEPEksQDEosxOELRHJwDNK2wggAPIMLQNvE7wWAKUY4ju6Q%2BAksDKKS77vyokJotaXUou8jEurEEcHhYKL%2BYsLbON8AYwAKAAangmD1Y7pxEwz/CCCIYjsFIMiCIoKjqL7uhcPohjGNY1j6HykuQCsVENMHEt1KiDQuJdUx%2BE3oThEMlQjE3RRZAIY96PPDTzHdeh2EPvQTG0ngdBvg/AwIfStGvM8JE3syTHv%2BSXzvZ%2BLFwKzBTTleMySWw8Lj%2BMi77ZMcKoAAHBuAAtBuSQexgDIGQHsPWgIRwQAplYSwTxcCEBeNiJuewPDK1VoaKET9eBM3dneUg7NObcz5lQzmAs8YcGFqQImJN/4SyljLEhpB5aIBAG%2Bc0BB1YQE1ngvQmB8BEAvtXYQohxBV1LvIJQahk4fzTqQRGadv4cAJow0W/9Hadj4c8byQDQHgMgdA2BXB4E%2BRwVrOSmClhENlmzDmXNqHUMFvQ3%2BzDxa2DYcQrQpC6FmC8bwFh7CAkrBRMQTIzhJBAA%3D

then we can keep the const variables which is nice.